From fb4d62fe77673bb11fe3e3ae716f58c9359ccfc1 Mon Sep 17 00:00:00 2001
From: Marvin Scholz <epirat07@gmail.com>
Date: Thu, 8 Oct 2020 15:26:55 +0200
Subject: [PATCH 26/26] demux/ogg: fix stream re-use

Stream re-use could never happen, as p_stream->p_es is always NULL
here, as implied by the earlier check:

   if ( p_stream->p_es == NULL && !p_stream->b_finished )

To fix that, the check for a non-null p_stream->p_es is removed.

Partially fixes #25186

(cherry picked from commit 3e039dd703e7e9d4a240432b1fa3fe4dfd045aee)
---
 modules/demux/ogg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/demux/ogg.c b/modules/demux/ogg.c
index ed7ab894c7..68c5dc1528 100644
--- a/modules/demux/ogg.c
+++ b/modules/demux/ogg.c
@@ -2205,7 +2205,7 @@ static void Ogg_CreateES( demux_t *p_demux )
             if( p_old_stream &&
                 p_old_stream->fmt.i_cat == p_stream->fmt.i_cat &&
                 p_old_stream->fmt.i_codec == p_stream->fmt.i_codec &&
-                p_old_stream->p_es != NULL && p_stream->p_es != NULL )
+                p_old_stream->p_es != NULL )
             {
                 msg_Dbg( p_demux, "will reuse old stream to avoid glitch" );
 
